azure-iot-device/azure/iot/device/common/pipeline/config.py

```diff
@@ -77,3 +84,23 @@ def _sanitize_cipher(cipher):
             raise TypeError("Invalid type for 'cipher'")
 
         return cipher
+
+    @staticmethod
+    def _rectify_keep_alive(keep_alive):
```

Comment:
>_rectify_keep_alive [](start = 8, length = 19)

This function just feels wrong to me.  _sanitize_cipher is needed because there is more than one way to specify cipher and we want to normalize it.  You're doing something different.  You're silently fixing out-of-bounds errors, and I can't think of may examples of that. 
Python talks about LBYL and EAFP and this is neither.  It's like you're adding a third option: LMFTFY (Let Me Fix That For You)